src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -51,29 +65,37 @@ class PythonToolRequirementsBase(Subsystem, ExportableTool):
 
     default_lockfile_resource: ClassVar[tuple[str, str] | None] = None
 
-    install_from_resolve = StrOption(
-        advanced=True,
-        default=None,
-        help=lambda cls: softwrap(
+    @classmethod
+    def _install_from_resolve_help(cls) -> str:
+        package_and_version = cls._default_package_name_and_version()
+        version_clause = (
+            f", which uses {package_and_version.name} version {package_and_version.version}"
```

Comment:
When this is formatted in the online docs, it might look better as inline code, hinting that it's referring to the PyPI name. What do you think?

Comparison: 

- as written: "... which uses isort version 23.3.0"
- suggestion: "... which uses `isort` version 23.3.0"

```suggestion
            f", which uses `{package_and_version.name}` version {package_and_version.version}"
```

Reply:
Good idea, I think this is definitely an improvement.